frontend/src/pages/list/POList.tsx

```diff
@@ -47,15 +46,190 @@ const COLUMNS: TableColumn[] = [
+  id: 0,
+  date: "",
+  vendorName: "",
+  uniqueBooks: 0,
+  totalBooks: 0,
+  totalCost: 0,
+};
+
+export default function PurchaseOrderList() {
+  // ----------------- STATE -----------------
+  const [loading, setLoading] = useState(false); // Whether we show that the table is loading or not
+  const [numberOfPurchaseOrders, setNumberOfPurchaseOrders] = useState(0); // The number of elements that match the query
+  const [purchaseOrders, setPurchaseOrders] = useState<PurchaseOrder[]>([]); // The data displayed in the table
+  const [deletePopupVisible, setDeletePopupVisible] = useState(false); // Whether the delete popup is shown
+  const [selectedDeletePurchaseOrder, setSelectedDeletePurchaseOrder] =
+    useState<PurchaseOrder>(emptyPurchaseOrder); // The element that has been clicked on to delete
+
+  // The current state of sorting.
+  const [sortParams, setSortParams] = useState<DataTableSortEvent>({
+    sortField: "",
+    sortOrder: null,
+    multiSortMeta: null, // Not used
+  });
+
+  // The current state of the paginator
+  const [pageParams, setPageParams] = useState<DataTablePageEvent>({
+    first: 0,
+    rows: NUM_ROWS,
+    page: 0,
+  });
+
+  // The current state of the filters
+  const [filterParams, setFilterParams] = useState<any>({
+    filters: {
+      id: { value: "", matchMode: "contains" },
+      date: { value: "", matchMode: "contains" },
+      vendorName: { value: "", matchMode: "contains" },
+      uniqueBooks: { value: "", matchMode: "contains" },
+      totalBooks: { value: "", matchMode: "contains" },
+      totalCost: { value: "", matchMode: "contains" },
+    } as Filters,
+  });
+
+  // ----------------- METHODS -----------------
+
+  // Callback functions for edit/delete buttons
+  const editPurchaseOrder = (po: PurchaseOrder) => {
+    console.log(po);
+  };
+
+  // Called to make delete pop up show
+  const deletePurchaseOrderPopup = (po: PurchaseOrder) => {
+    setSelectedDeletePurchaseOrder(po);
+    setDeletePopupVisible(true);
+  };
+
+  // Call to actually delete the element
+  const deletePurchaseOrderFinal = () => {
+    setDeletePopupVisible(false);
+    setSelectedDeletePurchaseOrder(emptyPurchaseOrder);
+  };
+
+  // Called when any of the filters (search boxes) are typed into
+  const onFilter = (event: DataTableFilterEvent) => {
+    setLoading(true);
+    setFilterParams(event);
+  };
+
+  // Called when any of the columns are selected to be sorted
+  const onSort = (event: DataTableSortEvent) => {
+    setLoading(true);
+    setSortParams(event);
+  };
+
+  // Called when the paginator page is switched
+  const onPage = (event: DataTablePageEvent) => {
+    setLoading(true);
+    setPageParams(event);
+  };
+
+  // When any of the list of params are changed, useEffect is called to hit the API endpoint
+  useEffect(() => callAPI(), [pageParams, sortParams, filterParams]);
+
+  // Calls the Vendors API
+  const callAPI = () => {
+    /*PURCHASES_API.getPurchaseOrders({
```

Comment:
Is this segment of code place holder code for when implementation of the API started?

Reply:
Yup, just a placeholder, as with below. This was crashing the app cause the endpoint does not exist yet.